From 82546b1f5fd5957c6d78f76488c0b947f3f8a508 Mon Sep 17 00:00:00 2001
From: Martin Guy <martinwguy@gmail.com>
Date: Sat, 13 Jul 2024 16:15:11 +0200
Subject: [PATCH] Fix signedness of spectrogram label position calculation

In spectrogram.c:870
        print_at(left + (p->cols - font_X * strlen(text)) / 2, 24,
Text, text);

strlen(txt) is unsigned, making the whole bracket expression unsigned
and in a calculating 4294967248/2 instead of -48.
---
 src/spectrogram.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/spectrogram.c b/src/spectrogram.c
index 89ae1fbf..f0cf4ca2 100644
--- a/src/spectrogram.c
+++ b/src/spectrogram.c
@@ -867,7 +867,7 @@ static int stop(sox_effect_t *effp) /* only called, by end(), on flow 0 */
     /* X-axis */
     step = axis(secs(p->cols), p->cols / (font_X * 9 / 2), &limit, &prefix);
     sprintf(text, "Time (%.1ss)", prefix);               /* Axis label */
-    print_at(left + (p->cols - font_X * strlen(text)) / 2, 24, Text, text);
+    print_at(left + (p->cols - font_X * (int)strlen(text)) / 2, 24, Text, text);
 
     for (i = 0; i <= limit; i += step) {
       int x = limit ? (double)i / limit * p->cols + .5 : 0;
-- 
2.39.2

